app/src/main/java/fr/neamar/kiss/DataHandler.java

```diff
@@ -191,22 +193,50 @@ public AppProvider getAppProvider() {
     ArrayList<Pojo> getFavorites(Context context, int limit) {
         ArrayList<Pojo> favorites = new ArrayList<>();
 
-        // Read history
-        ArrayList<ValuedHistoryRecord> ids = DBHelper.getFavorites(context, limit);
+        String favApps = PreferenceManager.getDefaultSharedPreferences(context).
+                getString("favorite-apps-list", "");
+        List<String> favAppsList = Arrays.asList(favApps.split(";"));
 
 
         // Find associated items
-        for (int i = 0; i < ids.size(); i++) {
-            Pojo pojo = getPojo(ids.get(i).record);
+        for (int i = 0; i < favAppsList.size(); i++) {
+            Pojo pojo = getPojo(favAppsList.get(i));
             if (pojo != null) {
                 favorites.add(pojo);
             }
         }
 
-
         return favorites;
     }
 
+    public boolean addToFavorites(Context context, String id) {
+
+        String favApps = PreferenceManager.getDefaultSharedPreferences(context).
+                getString("favorite-apps-list", "");
+        List<String> favAppsList = Arrays.asList(favApps.split(";"));
+        if (favAppsList.contains(id))
+        {
+            return false;
+        }
+
+        if (context instanceof MainActivity) {
```

Comment:
since this is the only case we're handling, maybe we can update the signature from Context directly to MainActivity?